File: app/src/main/java/com/arcgistest/di/components/ActivityComponent.java
package com.arcgistest.di.components;

import android.app.Activity;

import com.arcgistest.di.PerActivity;
import com.arcgistest.di.modules.ActivityModule;

import dagger.Component;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
    Activity activity();
}

File: app/src/main/java/com/arcgistest/di/components/ActivityComponent.kt
package com.arcgistest.di.components

import android.app.Activity

import com.arcgistest.di.PerActivity
import com.arcgistest.di.modules.ActivityModule

import dagger.Component

@PerActivity
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
    fun activity(): Activity
}
